### DISCUSSION TITLE
The X-Total-Count header is missing in the HTTP Response

### DISCUSSION THREAD

**Discussion by vaheeds** [hagopj13/node-express-boilerplate#102] [Q&A] [answered]
Hi,
I'm using this backend in combination with react-admin as frontend. Where do I need to modify to get rid of this error?

> The X-Total-Count header is missing in the HTTP Response. The jsonServer Data Provider expects responses for lists of resources to contain this header with the total number of results to build the pagination. If you are using CORS, did you declare X-Total-Count in the Access-Control-Expose-Headers header?

**Comment by fernandocanizo** [comment, marked as answer]
I haven't tested this, so try it and give us feedback please: you should add in the cors options the `X-Total-Count` in the `exposedHeaders` field.

Then to actually send the header with the page count, in the controller you should add a
```
res.setHeader("X-Total-Count", `${pageCount}`);
```

Also, I don't know about your `react-admin` thingy, but check if it accepts `Total-Count` as recommended by [RFC 6648](https://tools.ietf.org/html/rfc6648).

**Reply by vaheeds** [reply]
Thanks, I solved my issue ;like this:

` const result = await categoryService.queryCategories(filter, options);`
` res.append('X-Total-Count', result.totalResults);`
`  res.append('Access-Control-Expose-Headers', 'X-Total-Count');`
`  res.send(result.data);`
